Guides/Transactions.md

```diff
@@ -0,0 +1,111 @@
+
+**Note**: Transactions only work with MongoDB replica sets (v4.0+) and sharded clusters (v4.2+).
+
+## Examples
+
+### Transaction that Atomically Moves a `Document` from One `MongoCollection` to Another
+```swift
+let elg = MultiThreadedEventLoopGroup(numberOfThreads: 4)
+
+defer {
+    // free driver resources
+    client.syncShutdown()
+    cleanupMongoSwift()
+
+    // shut down EventLoopGroup
+    try? elg.syncShutdownGracefully()
+}
+
+let client = try MongoClient(using: elg)
+let session = client.startSession()
+
+let db = client.db("test")
+let srcColl = db.collection("src")
+let destColl = db.collection("coll")
+let docToMove: Document = ["hello": "world"]
+
+session.startTransaction().flatMap { _ in
+    srcColl.deleteOne(docToMove, session: session)
+}.flatMap { _ in
+    destColl.insertOne(docToMove, session: session)
+}.flatMap { _ in
+    session.commitTransaction()
+}.whenFailure { error in
+    session.abortTransaction()
+    // handle error
+}
+```
+
+### Transaction with Custom Transaction Options
+```swift
+let elg = MultiThreadedEventLoopGroup(numberOfThreads: 4)
+
+defer {
+    // free driver resources
+    client.syncShutdown()
+    cleanupMongoSwift()
+
+    // shut down EventLoopGroup
+    try? elg.syncShutdownGracefully()
+}
+
+let client = try MongoClient(using: elg)
+let session = client.startSession()
+
+let txnOpts = TransactionOptions(
+    maxCommitTimeMS: 30,
+    readConcern: ReadConcern(.local),
+    readPreference: ReadPreference.primaryPreferred,
+    writeConcern: try WriteConcern(w: .majority)
+)
+
+session.startTransaction(options: txnOpts).flatMap { _ in
+    // do something
+}.flatMap { _ in
+    session.commitTransaction()
+}.whenFailure { error in
+    session.abortTransaction()
+    // handle error
+}
+```
+
+### Transaction with Default Transaction Options
+```swift
+let elg = MultiThreadedEventLoopGroup(numberOfThreads: 4)
+
+defer {
+    // free driver resources
+    client.syncShutdown()
+    cleanupMongoSwift()
+
+    // shut down EventLoopGroup
+    try? elg.syncShutdownGracefully()
+}
+
+let txnOpts = TransactionOptions(
+    maxCommitTimeMS: 30,
+    readConcern: ReadConcern(.local),
+    readPreference: ReadPreference.primaryPreferred,
+    writeConcern: try WriteConcern(w: .majority)
+)
+
+let client = try MongoClient(using: elg)
+let session = client.startSession(options: ClientSessionOptions(defaultTransactionOptions: txnOpts))
+
+session.startTransaction().flatMap { _ in
+    // do something
+}.flatMap { _ in
+    session.commitTransaction()
+}.whenFailure { error in
+    session.abortTransaction()
+    // handle error
+}
+```
+
+Note: Any transaction options provided directly to `startTransaction()` override the default transaction options for the session. More so, the default transaction options for the session override any options inherited from the client.
```

Comment:
i think it would be a little clearer to move this example to before the `Transaction with Custom Transaction Options` one, and to add a blurb to the top of the `Transaction with Custom Transaction Options` one with this note.

Reply:
Done.